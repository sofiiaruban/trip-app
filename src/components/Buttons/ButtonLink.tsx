import { FC } from 'react'
import { Link } from 'react-router-dom'
import { ButtonType } from './constants'

interface ButtonLinkProps {
  url: string
  title: string
  type?: ButtonType
}

const ButtonLink: FC<ButtonLinkProps> = ({
  url,
  title,
  type = ButtonType.BUTTON
}) => {
  return (
    <Link to={url}>
      <button type={type}>{title}</button>
    </Link>
  )
}

export default ButtonLink
