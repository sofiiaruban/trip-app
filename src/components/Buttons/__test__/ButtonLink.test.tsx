import { BrowserRouter as Router } from 'react-router-dom'
import { render } from '@testing-library/react'
import ButtonLink from '../ButtonLink'

test('ButtonLink component matches snapshot', () => {
  const commonProps = {
    url: '/',
    title: 'Button Link'
  }

  const { container } = render(
    <Router>
      <ButtonLink {...commonProps} />
    </Router>
  )

  expect(container).toMatchSnapshot()
})
