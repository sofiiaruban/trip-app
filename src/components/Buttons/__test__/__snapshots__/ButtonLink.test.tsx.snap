// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonLink component matches snapshot 1`] = `
<div>
  <a
    href="/"
  >
    <button
      type="button"
    >
      Button Link
    </button>
  </a>
</div>
`;
